test TestAceValidator fails

Sometimes, the test TestAceValidator fails. For example on [Semaphore](https://semaphoreci.com/coreos/rkt/branches/pull-request-1780/builds/3):

```
--- FAIL: TestAceValidator (6.95s)
	rkt_tests.go:111: Running command: /home/runner/rkt/builds/build-rkt-coreos-/build-rkt-0.11.0+git/bin/rkt --dir=/tmp/datadir-805339672 --local-config=/tmp/localdir-033251479 --system-config=/tmp/systemdir-983410442 --debug --insecure-skip-verify run --mds-register --volume database,kind=empty /home/runner/rkt/builds/build-rkt-coreos-/build-rkt-0.11.0+git/tmp/functional/rkt-ace-validator-main.aci /home/runner/rkt/builds/build-rkt-coreos-/build-rkt-0.11.0+git/tmp/functional/rkt-ace-validator-sidekick.aci
	rkt_ace_validator_test.go:58: Expected "sidekick OK" but not found: rkt: using image from file /home/runner/rkt/builds/build-rkt-coreos-/build-rkt-0.11.0+git/bin/stage1-coreos.aci

...

		[  243.210358] ace-validator[6]: sidekick OK

```

The command prints "sidekick OK" correctly but gexpect does not always catch it for some reasons.

--- a/tests/rkt_ace_validator_test.go
+++ b/tests/rkt_ace_validator_test.go
@@ -16,7 +16,6 @@ package main
 
 import (
 	"fmt"
-	"os"
 	"testing"
 
 	"github.com/coreos/rkt/tests/testutils"
@@ -37,14 +36,8 @@ func TestAceValidator(t *testing.T) {
 		t.Fatalf("Cannot launch metadata service: %v", err)
 	}
 
-	aceMain := os.Getenv("RKT_ACE_MAIN_IMAGE")
-	if aceMain == "" {
-		panic("empty RKT_ACE_MAIN_IMAGE env var")
-	}
-	aceSidekick := os.Getenv("RKT_ACE_SIDEKICK_IMAGE")
-	if aceSidekick == "" {
-		panic("empty RKT_ACE_SIDEKICK_IMAGE env var")
-	}
+	aceMain := testutils.GetValueFromEnvOrPanic("RKT_ACE_MAIN_IMAGE")
+	aceSidekick := testutils.GetValueFromEnvOrPanic("RKT_ACE_SIDEKICK_IMAGE")
 
 	rktArgs := fmt.Sprintf("--debug --insecure-skip-verify run --mds-register --volume database,kind=empty %s %s",
 		aceMain, aceSidekick)

--- a/tests/rkt_ace_validator_test.go
+++ b/tests/rkt_ace_validator_test.go
@@ -16,19 +16,27 @@ package main
 
 import (
 	"fmt"
+	"strings"
 	"testing"
 
 	"github.com/coreos/rkt/tests/testutils"
 )
 
-var expectedResults = []string{
-	"prestart OK",
-	"main OK",
-	"sidekick OK",
-	"poststop OK",
-}
-
 func TestAceValidator(t *testing.T) {
+	newStringSet := func(strs ...string) map[string]struct{} {
+		m := make(map[string]struct{}, len(strs))
+		for _, s := range strs {
+			m[s] = struct{}{}
+		}
+		return m
+	}
+	expected := []map[string]struct{}{
+		newStringSet("prestart"),
+		newStringSet("main", "sidekick"),
+		newStringSet("poststop"),
+	}
+	pattern := `ace-validator\[\d+\]: ([[:alpha:]]+) OK`
+
 	ctx := testutils.NewRktRunCtx()
 	defer ctx.Cleanup()
 
@@ -46,9 +54,27 @@ func TestAceValidator(t *testing.T) {
 	child := spawnOrFail(t, rktCmd)
 	defer waitOrFail(t, child, true)
 
-	for _, e := range expectedResults {
-		if err := expectWithOutput(child, e); err != nil {
-			t.Fatalf("Expected %q but not found: %v", e, err)
+	for _, set := range expected {
+		for len(set) > 0 {
+			results, _, err := expectRegexWithOutput(child, pattern)
+			if err != nil {
+				var keys []string
+				for k := range set {
+					keys = append(keys, fmt.Sprintf("%q", k))
+				}
+				ex := strings.Join(keys, " or ")
+				t.Fatalf("Expected %s, but not found: %v", ex, err)
+			}
+			if len(results) != 2 {
+				t.Fatalf("Unexpected regex results, expected a whole match and one submatch, got %#v", results)
+			}
+			aceStage := results[1]
+			if _, ok := set[aceStage]; ok {
+				t.Logf("Got expected ACE stage %q", aceStage)
+				delete(set, aceStage)
+			} else {
+				t.Logf("Ignoring unknown ACE stage %q", aceStage)
+			}
 		}
 	}
 }

